src/python/pants/backend/scala/test/scalatest.py

```diff
@@ -141,6 +142,7 @@ async def setup_scalatest_for_target(
         extra_immutable_input_digests=extra_immutable_input_digests,
         output_directories=(reports_dir,),
         description=f"Run Scalatest runner for {request.field_set.address}",
+        timeout_seconds=request.field_set.timeout.value,
```

Comment:
On this first stab at this feature I decided to pass the timeout value directly into the `JvmProcess` and rely on the new defaults feature for global default values.

If an approach more akin to how `pytest` does is desired (allowing for a global max timeout at the subsystem) just let me know and I can proceed adding that.

Reply:
I like the idea of using a per-target option like you have done here, but I think a global default will indeed be necessary.

Reply:
Hm, yea: this is tricky. I think that "timeouts by default" is a good idea, so building in a global default in an option might make sense. Probably worth doing in this PR.

It's also possible that now that we have `__defaults__` to allow for overriding at the target-type level, the existing `pytest` timeout options (to set a default, and to enable/disable enforcement) could be deprecated, and moved to the `test` `goal` itself. To do that, you'd deprecate the old option field by setting a `removal_version='2.15.0.dev1'`, and consume from both old/new options with https://github.com/pantsbuild/pants/blob/032985ac7393754182cc6a07e8d7312efa7a1972/src/python/pants/base/deprecated_test.py#L207-L241. If you don't have time to do that here though, that's fine.

Reply:
I'm more inclined to move the code towards incentivating the usage of the new `__defaults__` construct (so, deprecating the `pytest` subsystem options) as that seems to give a more uniform approach to set defaults than learning what are the specific options for each of the subsystems...

However, since the `pytest` subsystem has `max_timeout` option, I understand that we would keep that one around as it has no equivalent as a target field.